retrieval/discovery/kubernetes/endpoints.go

```diff
@@ -89,6 +89,11 @@ func (e *Endpoints) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			send(e.buildEndpoints(o.(*apiv1.Endpoints)))
 		},
 		DeleteFunc: func(o interface{}) {
+			s, ok := o.(*cache.DeletedFinalStateUnknown)
+			if ok {
+				o = s.Obj
+			}
+
 			send(&config.TargetGroup{Source: endpointsSource(o.(*apiv1.Endpoints).ObjectMeta)})
```

Comment:
Is it worth adding a type assertion test to make sure there's no chance of a panic here?